.github/workflows/configurable-benchmark-test.yaml

```diff
@@ -1,64 +1,69 @@
-        required: false
-      service_mesh:
-        type: choice
-        required: false
-        description: "service mesh being tested"
-        options:
-          - istio
-          - linkerd
-      load_generator:
-        type: choice
-        required: false
-        description: "load generator to run tests with"
-        options:
-          - fortio
-          - wrk2
-          - nighthawk
-
-jobs:
-  manual-test:
-    name: Configurable Benchmark Test
-    runs-on: ubuntu-latest
-    if: ${{ github.event_name == 'workflow_dispatch' }}
-    steps:
-      - name: Setup Kubernetes
-        uses: manusa/actions-setup-minikube@v2.4.3
-        with:
-          minikube version: 'v1.23.2'
-          kubernetes version: 'v1.23.2'
-          driver: docker
-          start args: "--v=5"
-
-      - name: Checkout Code
-        uses: actions/checkout@v2
-
-      - name: Install Service Mesh and Deploy Application
-        run: |
-          chmod +x .github/workflows/scripts/${{ github.event.inputs.service_mesh }}_deploy.sh
-          .github/workflows/scripts/${{ github.event.inputs.service_mesh }}_deploy.sh
-        shell: bash
-        
-      - name: Run Benchmark Tests
-        uses: layer5io/meshery-smp-action@master
-        with:
-          provider_token: ${{ secrets.MESHERY_TOKEN }}
-          platform: docker
-          profile_name: ${{ github.event.inputs.profile_name }}
-          profile_filename: ${{ github.event.inputs.profile_filename }}
-          endpoint_url: ${{env.ENDPOINT_URL}}
-          service_mesh: ${{env.SERVICE_MESH}}
-          load_generator: ${{ github.event.inputs.load_generator }}
-          test_name: '${{ github.event.inputs.service_mesh }}-${{ github.event.inputs.load_generator }}-${{ github.event.inputs.profile_filename }}${{ github.event.inputs.profile_name }}'
+# This workflow runs performance benchmarks with Meshery based on the configuration provided
+# This workflow needs to be triggered manually by providing the test configuration
+
+name: Configurable Benchmark Test
+on:
+  # for triggering the workflow, provide these inputs
+  workflow_dispatch:
+    inputs:
+      profile_name:
+        description: "performance profile to use"
+        required: false
+      profile_filename:
+        description: "test configuration file"
+        required: false
+      service_mesh:
+        type: choice
+        required: false
+        description: "service mesh being tested"
+        options:
+          - istio
+          - linkerd
+          - osm
+      load_generator:
+        type: choice
+        required: false
+        description: "load generator to run tests with"
+        options:
+          - fortio
+          - wrk2
+          - nighthawk
+
+jobs:
+  manual-test:
+    name: Configurable Benchmark Test
+    runs-on: ubuntu-latest
+    if: ${{ github.event_name == 'workflow_dispatch' }}
+    steps:
+      - name: Setup Kubernetes
+        uses: medyagh/setup-minikube@v0.0.7

```

Comment:
@hershd23 Can we still use manusa/actions-setup-minikube@v2.4.3 and use `start args: "--cpu 4 --memory 5192"` to set the cpu and memory? Because it's better to reduce unnecessary changes in this PR. I suggested it in #48, can you try to prove that everything when using `manusa/actions-setup-minikube` ?

Reply:
Sure

Reply:
Thank you. @hershd23

Reply:
https://github.com/hershd23/meshery-smp-action/runs/7053416986?check_suite_focus=true

Works fine, I had to increase the sleep timer for istio though

Reply:
Great!